internal/app/init/pkg/system/system.go

```diff
@@ -138,3 +138,28 @@ func (s *singleton) List() (result []*ServiceRunner) {
 
 	return
 }
+
+// Stop will initiate a shutdown of the specified service.
+func (s *singleton) Stop(services ...Service) {
+	if len(services) == 0 {
+		return
+	}
+
+	s.mu.Lock()
+	if s.terminating {
+		s.mu.Unlock()
+		return
+	}
+
+	stateCopy := make(map[string]*ServiceRunner)
+	for _, svc := range services {
+		id := svc.ID(s.UserData)
+		stateCopy[id] = s.State[id]
+	}
+	s.mu.Unlock()
+
+	for id, svc := range stateCopy {
```

Comment:
nitpicking here, but `svc` is actually `ServiceRunner`, probably it makes sense to have a variable named `svcrunner` or something similar?